book/src/qobject/signals_enum.md

```diff
@@ -7,18 +7,31 @@ SPDX-License-Identifier: MIT OR Apache-2.0
 
 # Signals enum
 
-The `cxx_qt::qsignals(T)` attribute is used on an enum to define which signals should exist on the QObject `T`. It allows you to define the signal name and the parameters of the signal.
+The `cxx_qt::qsignals(T)` attribute is used on an [enum](https://doc.rust-lang.org/book/ch06-01-defining-an-enum.html) to define [signals](https://doc.qt.io/qt-6/signalsandslots.html) for the QObject `T`.
 
 ```rust,ignore,noplayground
 {{#include ../../../examples/qml_features/rust/src/signals.rs:book_signals_enum}}
 ```
 
+For every enum variant CXX-Qt will generate a signal on the corresponding QObject.
+If the enum variant has members, they will become the parameters for the corresponding signal.
+
+Because CXX-Qt needs to know the names of each parameter, only enum variants with named members are supported.
+The Signal parameters are generated in order of appearance in the enum variant.
+
 ## Emitting a signal
 
-To emit a signal from Rust use the [`CppObj`](./cpp_object.md) and call either the `emit_queued(Signal)` method.
+For every generated QObject [`qobject::T`](./generated-qobject.md) that has a signals enum CXX-Qt will generate an `emit` function:
+``` rust,ignore,noplayground
+fn emit(self: Pin<&mut Self>, signal: /*Signals enum*/)
+```
+`emit` can therefore be called from any mutable `#[qinvokable]`.
 
-Note that signals are queued to avoid a deadlock if the `Q_EMIT` is `Qt::DirectConnection` connected to a Rust invokable on the same QObject that has caused the `Q_EMIT`, as this would then try to lock the internal Rust object which is already locked.
+The `emit` function will immediately emit the signal and call any connected slots.
```

Comment:
This is confusing. Is CXX-Qt calling the slots directly or is it just a Qt::DirectConnection? Linking to https://doc.qt.io/qt-6/qt.html#ConnectionType-enum may be helpful here.

Reply:
It's just calling `Q_EMIT mySignal();` on the C++ side in the same way you would in C++, then depending how other things are connected to that signal depends what connection type is used to call the slots.

Reply:
Wording should be clearer now :)